include/ceed.h

```diff
@@ -162,15 +162,12 @@ CEED_EXTERN int CeedErrorImpl(Ceed, const char *, int, const char *, int,
 ///
 /// @ingroup Ceed
 /// @sa CeedSetErrorHandler()
-#if defined(__clang__)
-/// Use nonstandard ternary to convince the compiler/clang-tidy that this
-/// function never returns zero.
 #  define CeedError(ceed, ecode, ...)                                     \
-  (CeedErrorImpl((ceed), __FILE__, __LINE__, __func__, (ecode), __VA_ARGS__) ?: (ecode))
-#else
-#  define CeedError(ceed, ecode, ...)                                     \
-  CeedErrorImpl((ceed), __FILE__, __LINE__, __func__, (ecode), __VA_ARGS__) ?: (ecode)
-#endif
+  CeedErrorImpl((ceed), __FILE__, __LINE__, __func__, (ecode), __VA_ARGS__) ? 0 : (ecode)
```

Comment:
Why these changes?

Reply:
I suspect this may be part of why `make tidy` is unhappy.

Reply:
Hmm, can we leave the clang alternative and change the `#else` part to drop the ternary? (I think that was a mistake in the original commit.)

Reply:
Parts of the code use `CeedError` without using its value which causes some compilation flags to print warnings. I split it into `CeedError` and `CeedPrintError` to handle the non-return cases.

Reply:
It's probably sufficient to `(void)CeedError(...);` in that one place where the value isn't used. In any case, the `0` is not the desired semantic.